packages/angular_devkit/core/src/virtual-fs/host/memory.ts

```diff
@@ -193,29 +231,105 @@ export class SimpleMemoryHost implements Host<{}> {
       }
     } else {
       for (const p of this._cache.keys()) {
-        if (p.startsWith(NormalizedSep)) {
+        if (p.startsWith(NormalizedSep) && p !== NormalizedRoot) {
           result.add(split(p)[1]);
         }
       }
     }
 
-    return observableOf([...result]);
+    return [...result];
   }
 
-  exists(path: Path): Observable<boolean> {
-    path = this._toAbsolute(path);
+  _exists(path: Path): boolean {
+    return !!this._cache.get(this._toAbsolute(path));
+  }
+  _isDirectory(path: Path): boolean {
+    const maybeStats = this._cache.get(this._toAbsolute(path));
+
+    return maybeStats ? maybeStats.isDirectory() : false;
+  }
+  _isFile(path: Path): boolean {
+    const maybeStats = this._cache.get(this._toAbsolute(path));
+
+    return maybeStats ? maybeStats.isFile() : false;
+  }
+
+  _stat(path: Path): Stats<SimpleMemoryHostStats> {
+    const maybeStats = this._cache.get(this._toAbsolute(path));
+
+    if (!maybeStats) {
+      throw new FileDoesNotExistException(path);
+    } else {
+      return maybeStats;
+    }
+  }
+
+  write(path: Path, content: FileBuffer): Observable<void> {
+    return new Observable<void>(obs => {
+      this._write(path, content);
+      obs.next();
+      obs.complete();
+    });
+  }
+
+  read(path: Path): Observable<FileBuffer> {
+    return new Observable<FileBuffer>(obs => {
+      const content = this._read(path);
+      obs.next(content);
+      obs.complete();
+    });
+  }
+
+  delete(path: Path): Observable<void> {
+    return new Observable<void>(obs => {
+      this._delete(path);
+      obs.next();
+      obs.complete();
+    });
+  }
+
+  rename(from: Path, to: Path): Observable<void> {
+    return new Observable<void>(obs => {
+      this._rename(from, to);
+      obs.next();
+      obs.complete();
+    });
+  }
 
-    return observableOf(this._cache.has(path) || this._isDir(path));
+  list(path: Path): Observable<PathFragment[]> {
```

Comment:
Thoughts on this returning `Observable<PathFragment>` instead of the array.  Would allow the consumer to operator on the stream of elements directly.

Reply:
I see `Observable<X>` here as a replacement of `Promise<>`. The contract is that these will always emit once and complete, or error out. I like the consistence of this interface too.